spikeextractors/extractors/neoextractors/neobaseextractor.py

```diff
@@ -78,28 +78,42 @@ def __init__(self, block_index=None, seg_index=None, **kargs):
         self.additional_gain[units == 'uV'] = 1.
         self.additional_gain = self.additional_gain.reshape(1, -1)
 
+        # Add channels properties
+        header_channels = self.neo_reader.header['signal_channels'][slice(None)]
+        channel_ids = self.get_channel_ids()
+
+        gains = header_channels['gain'] * self.additional_gain[0]
+        self.set_channel_gains(gains=gains, channel_ids=channel_ids)
+        
+        names = header_channels['name']
+        for i, ind in enumerate(channel_ids):
+            self.set_channel_property(channel_id=ind, property_name='name', value=names[i])
+
     @check_get_traces_args
-    def get_traces(self, channel_ids=None, start_frame=None, end_frame=None):
+    def get_traces(self, channel_ids=None, start_frame=None, end_frame=None, return_scaled=True):
         # in neo rawio channel can acces by names/ids/indexes
         # there is no garranty that ids/names are unique on some formats
         raw_traces = self.neo_reader.get_analogsignal_chunk(block_index=self.block_index, seg_index=self.seg_index,
                                                             i_start=start_frame, i_stop=end_frame,
                                                             channel_indexes=None, channel_names=None,
                                                             channel_ids=channel_ids)
 
-        # rescale traces to natural units (can be anything)
-        scaled_traces = self.neo_reader.rescale_signal_raw_to_float(raw_traces, dtype='float32',
-                                                                    channel_indexes=None, channel_names=None,
-                                                                    channel_ids=channel_ids)
-        channel_idxs = np.array([list(channel_ids).index(ch) for ch in channel_ids])
-        # and then to uV
-        scaled_traces *= self.additional_gain[:, channel_idxs]
+        if return_scaled:
+            # rescale traces to natural units (can be anything)
+            scaled_traces = self.neo_reader.rescale_signal_raw_to_float(raw_traces, dtype='float32',
+                                                                        channel_indexes=None, channel_names=None,
+                                                                        channel_ids=channel_ids)
+            channel_idxs = np.array([list(channel_ids).index(ch) for ch in channel_ids])
+            # and then to uV
+            scaled_traces *= self.additional_gain[:, channel_idxs]
 
-        # fortunatly neo works with (samples, channels) strides
-        # so transpose to spieextractors wolrd
-        scaled_traces = scaled_traces.transpose()
+            # fortunatly neo works with (samples, channels) strides
+            # so transpose to spieextractors wolrd
+            scaled_traces = scaled_traces.transpose()
```

Comment:
scaled_traces = scaled_traces.transpose() should not be in "if return_scaled:"

Reply:
fixed